File: chattools/src/main/java/aytackydln/chattools/telegram/dto/models/CustomKeyboard.java
package aytackydln.chattools.telegram.dto.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@Data
@RequiredArgsConstructor
public class CustomKeyboard implements Keyboard {
    private final List<List<KeyboardItem>> keyboard = new ArrayList<>();

    @JsonProperty
    public boolean isResizeKeyboard() {
        return true;
    }

    @JsonProperty
    public boolean isOneTimeKeyboard() {
        return true;
    }

    @JsonProperty
    public List<List<KeyboardItem>> getKeyboard() {
        return keyboard;
    }

    public List<KeyboardItem> addRow(){
        List<KeyboardItem> newList=new ArrayList<>();
        keyboard.add(newList);
        return newList;
    }
}

File: chattools/src/main/kotlin/aytackydln/chattools/telegram/dto/models/CustomKeyboard.kt
package aytackydln.chattools.telegram.dto.models

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming
import lombok.Data
import lombok.RequiredArgsConstructor

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
class CustomKeyboard : Keyboard {
    @get:JsonProperty
    val keyboard: MutableList<MutableList<KeyboardItem>> = ArrayList()

    @get:JsonProperty
    val isResizeKeyboard: Boolean
        get() = true

    @get:JsonProperty
    val isOneTimeKeyboard: Boolean
        get() = true

    fun addRow(): MutableList<KeyboardItem> {
        val newList: MutableList<KeyboardItem> = ArrayList()
        keyboard.add(newList)
        return newList
    }
}
